### PR TITLE
feat: add alternative TorchToTosa conversion for AtenGeluOp.

### PR DESCRIPTION
**Pull request by ttjost**

### CODE REVIEW

File: test/Conversion/TorchToTosa/basic.mlir [hunk 1]
```diff
@@ -1134,3 +1134,51 @@ func.func @torch.aten.remainder.Scalar(%arg0: !torch.vtensor<[2, 4],f32>) -> !to
   %0 = torch.aten.remainder.Scalar %arg0, %int2 : !torch.vtensor<[2, 4],f32>, !torch.int -> !torch.vtensor<[2, 4],f32>
   return %0 : !torch.vtensor<[2, 4],f32>
 }
+
```
Comment (ttjost) [inline test/Conversion/TorchToTosa/basic.mlir:1137]: @mgehre-amd , I guess the normal flow already generated `aten.gelu`. Would you say we don't need this test?

Comment (mgehre-amd) [reply test/Conversion/TorchToTosa/basic.mlir:1137]: I suggest to revert the changes to this file. The normal gelu lowering should be tested via end2end tests already.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 2]
```diff
@@ -5524,6 +5524,48 @@ class ConvertAtenOpToTosaCustomOp : public OpConversionPattern<AtenOpT> {
...
+                              std::string /*implementedWithOpAttr*/)
+      : OpConversionPattern<AtenGeluOp>(typeConverter, context) {}
+
+  LogicalResult
+  matchAndRewrite(AtenGeluOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    std::string approximate;
+    if (!matchPattern(op.getApproximate(), m_TorchConstantStr(approximate)) ||
+        approximate != "none") {
```
Comment (mgehre-amd) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:5545]: Why do we care whether approximate is `none` here? Can we forward all values?

**Review by mgehre-amd — APPROVED** [review]

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 2 above)
Comment (ttjost) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:5545]: You are right. Removing it.

Comment (ttjost) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:5545]: That came from the original conversion, and I hadn't thought it through. Nice catch!

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 3]
```diff
@@ -5524,6 +5524,47 @@ class ConvertAtenOpToTosaCustomOp : public OpConversionPattern<AtenOpT> {
...
+template <>
+class ConvertAtenOpToTosaCustomOp<AtenGeluOp>
+    : public OpConversionPattern<AtenGeluOp> {
+public:
+  using OpConversionPattern<AtenGeluOp>::OpConversionPattern;
+  using OpAdaptor = typename AtenGeluOp::Adaptor;
+
+  ConvertAtenOpToTosaCustomOp(TypeConverter &typeConverter,
+                              MLIRContext *context, std::string /*opName*/,
+                              std::string /*implementedWithOpAttr*/)
```
Comment (TinaAMD) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:5536]: Isn't `implementedWithOpAttr` already `"UNDEF"` and you can use this later on for `implement_attrs`?

File: test/Conversion/TorchToTosa/basic.mlir [hunk 4]
```diff
@@ -1133,4 +1133,4 @@ func.func @torch.aten.remainder.Scalar(%arg0: !torch.vtensor<[2, 4],f32>) -> !to
   %int2 = torch.constant.int 2
   %0 = torch.aten.remainder.Scalar %arg0, %int2 : !torch.vtensor<[2, 4],f32>, !torch.int -> !torch.vtensor<[2, 4],f32>
   return %0 : !torch.vtensor<[2, 4],f32>
-}
+}
```
Comment (TinaAMD) [inline test/Conversion/TorchToTosa/basic.mlir:1136]:
```suggestion
}

```

**Review by TinaAMD — APPROVED** [review]

File: test/Conversion/TorchToTosa/basic.mlir (hunk 4 above)
Comment (ttjost) [reply test/Conversion/TorchToTosa/basic.mlir]: Yes, this is going to be removed.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 5]
```diff
@@ -5951,15 +5993,19 @@ class ConvertTorchToTosa : public ConvertTorchToTosaBase<ConvertTorchToTosa> {
...
+        typeConverter, context, opName, implementedWith);                      \
+  }
     INSERT_ATEN_TO_TOSA_CUSTOMOP_PATTERN(AtenAtan2Op, "math.atan2",
                                          "linalg.generic");
     INSERT_ATEN_TO_TOSA_CUSTOMOP_PATTERN(AtenSinOp, "math.sin",
                                          "linalg.generic");
     INSERT_ATEN_TO_TOSA_CUSTOMOP_PATTERN(AtenCosOp, "math.cos",
                                          "linalg.generic");
+    if (this->enableCustomOpConversion)
+      INSERT_ATEN_TO_TOSA_CUSTOMOP_PATTERN(AtenGeluOp, "UNDEF", "UNDEF");
```
[the remark this answers is not in the document]
Comment (ttjost) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:6008]: So, I am changing to `NOT_USED` to explicitly tell that we don't use it.  =)

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 3 above)
Comment (ttjost) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:5536]: We would need to store it somewhere, but for this specific customOp gelu, we don't the Conversion pass to store anything.  =)

**Comment by ttjost** [comment]
Closing this PR since this has removed to another repository.

**Pull request closed without merging**